src/js/components/utils/editModal.tsx

```diff
@@ -0,0 +1,125 @@
+import Button from '@salesforce/design-system-react/components/button';
+import Input from '@salesforce/design-system-react/components/input';
+import Modal from '@salesforce/design-system-react/components/modal';
+import Textarea from '@salesforce/design-system-react/components/textarea';
+import i18n from 'i18next';
+import { omit } from 'lodash';
+import React, { useEffect, useRef } from 'react';
+
+import { useForm, useIsMounted } from '@/components/utils';
+import { Project } from '@/store/projects/reducer';
+import { OBJECT_TYPES } from '@/utils/constants';
+
+interface EditModalProps {
+  project: Project;
+  isOpen: boolean;
+  handleClose: () => void;
+}
+const EditModal = ({ project, isOpen, handleClose }: EditModalProps) => {
+  const isMounted = useIsMounted();
+  const submitButton = useRef<HTMLButtonElement | null>(null);
+
+  const handleSuccess = () => {
+    /* istanbul ignore else */
+    if (isMounted.current) {
+      handleClose();
+    }
+  };
+
+  const defaultName = project.name;
+  const defaultDescription = project.description.replace(/<(.|\n)*?>/g, '');
```

Comment:
Oh ho -- interesting. This is not ideal, and I didn't think about it until now.

@wlonk So, we are currently _only_ exposing the parsed markdown to the front end for project/task descriptions. That doesn't work great for editing existing descriptions... In an ideal world (is that where we live?), the front end could always submit unparsed markdown, but have access to both parsed (for display) and unparsed (for editing). What do you think?

Reply:
I think that's very doable. I would normally do that with a pair of fields, as we've done before:
 - `foo` (for the raw markdown, r/w field)
 - `foo_rendered` (for the parsed markdown, ro field)

Reply:
@jgerigmeyer is that part of this story, or should it be its own?

Reply:
@wlonk I think it really needs to be part of this story/PR.

Reply:
Cool, happy to do it here, but I didn't want to start it here and slow this PR down if you had meant for it to be in a different PR!